### ISSUE TITLE
DBBrowser compare types shows wrong color for negative values where lower is better

### ISSUE THREAD

**Issue by wffurr** [osmium-org/osmium#101]
e.g. when comparing resists on shield hardeners, the colors are wrong for the resist column:

https://o.smium.org/db/comparetypes/2295,9660,2303,20635,13995,17497,13966,19261,19262,19268,19267,19278,19277,19284,19283,14760,14761,14762,14763/auto

![screen shot 2015-06-05 at 3 06 44 pm](https://cloud.githubusercontent.com/assets/519506/8013448/37e4bd10-0b97-11e5-9bc7-92abba83156a.png)

The problem is with negative values where lower values (e.g. more negative) are better.  -55% is better than -50%, etc.

The delta is calculated as the difference.  So in the screenshot, where -50% is the base value and -55% is the value we're trying to color, -55% is 10% more shield resist, so the delta = 10.

The text comes out correct as +10%.

However, the color ends up red instead of green.  Shield resist is correctly marked with "hig" (high is good) of 0.  Smaller, aka more negative, shield resist values are better.  However, the delta is positive, so the conditional adds the "loss" class instead of the "gain" class.

I think the gain var at line 126 should be absolute value.  Then it would represent "gaining" shield resist as well as it represents "gaining" volume or CPU usage.

https://github.com/osmium-org/osmium/blob/staging/src/snippets/dbbrowser.js#L126

**Comment by Artefact2** [comment]
The problem is that most attributes have a nonsensical highisgood value in the game database. I have hardcoded a few of them [in the import script](https://github.com/osmium-org/osmium/blob/v0.13.4-hotfixes/bin/reverence_insert#L116) but of course some attributes still slip through.

**Comment by Artefact2** [comment]
Fixed in 8a046a7.

**Comment by wffurr** [comment]
Thermal Damage Resistance Bonus (987) already had hig=0.  That's not the problem here.  The problem is with comparing negative values for a field with hig=0.  delta comes out as a positive number, even though the better field is *more negative*.

e.g.
```javascript
var baseval = -50;
var otherval = -55;
var delta = 100 * (otherval - baseval) / baseval; // delta == 10
var highisgood = 0;
var gain = delta >= 0;  // gain == true
if(highisgood) {  // false so go to else
  addClass(gain ? 'gain' : 'loss');
} else {
  addClass(gain ? 'loss' : 'gain');  // gain == true, so add "loss", oops!
}
```

We should get `gain` since we have 10% better (aka more negative) shield resist, but instead we get `loss`.

If you had:

```javascript
var delta = 100 * (otherval - baseval) / Math.abs(baseval);
```

Then `delta` is correct for both positive and negative values.  Using Math.abs in the denominator makes the baseval be just a normalization factor to give us the percentage difference.

baseval | otherval | delta
-------- | -------- | -----
50 | 55 | 10
55 | 50 | -9.1
-50 | -55 | -10
-55 | -50 | 9.1

**Comment by Artefact2** [comment]
You are right. I was fooled by the [attribute page](https://o.smium.org/db/attribute/987) saying highisgood is true, when it's not !

**Comment by Artefact2** [comment]
Fixed (for good this time!) in 94146c3ea81e4bd5ecc3a29015ccb5434b0dd299.
